### PR TITLE
Wk 233 stadsdeel filter

### PR DESCRIPTION
**Pull request by aditudorache**
This PR adds filter functionality for the stadsdeel in the map. Also enables the call of the export functionality with stadsdeel filter and for all data. Unittests and refactoring of the filter component iare not included.

### CODE REVIEW

File: src/components/filterPanel/FilterPanel.js [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+
+const ExportButton = styled(Button)`
+  margin: ${themeSpacing(2, 0)};
+`;
+
+const FilterWrapperStyle = styled.div`
+  max-height: calc(780px - 33px);
+  overflow-x: hidden;
+  overflow-y: auto;
+  will-change: transform;
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: I don't see a `transform` style rule. Is this line necessary?

File: src/components/filterPanel/FilterPanel.js [hunk 2]
```diff
@@ -25,24 +39,61 @@ function getStatusClassName(status) {
   return statusClassMapper[status];
 }
 
+const exportUrl = `${endpoints.blackspotsExport}`;
+
+export const getExportFilter = stadsdeelFilter => {
+  if (Object.values(stadsdeelFilter).filter(e => e).length === 0) return '';
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: Or
```javascript
Object.values(stadsdeelFilter).filter(Boolean)
```

File: src/components/filterPanel/FilterPanel.js [hunk 3]
```diff
@@ -25,24 +39,61 @@ function getStatusClassName(status) {
...
   const { trackEvent } = useMatomo();
+  const [downloadUrl, setDownloadUrl] = useState(exportUrl);
+  const [canDownload, setCanDownload] = useState(true);
 
-  const trackFilter= (name) => {
-    trackEvent({category: 'Map filters', action: name});
-  }
+  const [, downloadFile] = useDownload();
+
+  const exportFilter = () => {
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: Wrap this in a `useCallback`?

File: src/components/filterPanel/FilterPanel.js [hunk 4]
```diff
@@ -25,24 +39,61 @@ function getStatusClassName(status) {
...
+  useEffect(() => {
+    setDownloadUrl(`${exportUrl}${getExportFilter(stadsdeelFilter)}`);
+    setCanDownload(
+      Object.values(stadsdeelFilter).filter(e => e).length <= 1 &&
+        Object.values(spotTypeFilter).filter(e => e).length === 0 &&
+        Object.values(spotStatusTypeFilter).filter(e => e).length === 0 &&
+        optionValue === ContextMenuOptions.ALL
+    );
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [stadsdeelFilter]);
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: Dependencies should include `spotTypeFilter` and `spotStatusTypeFilter`

File: src/components/filterPanel/FilterPanel.js [hunk 5]
```diff
@@ -25,24 +39,61 @@ function getStatusClassName(status) {
...
+
+  const trackFilter = name => {
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: Wrap this in a `useMemo`?

File: src/components/filterPanel/FilterPanel.js [hunk 6]
```diff
@@ -344,6 +379,45 @@ const FilterPanel = ({
     );
   }
 
+  const renderStadsdeelCheckboxes = () => {
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: `useMemo`?

File: src/components/filterPanel/FilterPanel.js [hunk 7]
```diff
@@ -360,31 +434,38 @@ const FilterPanel = ({
...
-          : ''}
-
-        {optionValue !== ContextMenuOptions.DELIVERED
-          ? renderStatusCheckboxes()
-          : ''}
-      </div>
+      <FilterWrapperStyle>
+        <div className={styles.FilterContainer}>
+          {renderOptions()}
+          {optionValue !== ContextMenuOptions.ALL ? <h5>Jaar</h5> : ''}
```
Comment (janjaap) [inline src/components/filterPanel/FilterPanel.js]: Conditional rendering should return a component or `undefined` or `null`.
You can write
```jsx
{optionValue !== ContextMenuOptions.ALL && <h5>Jaar</h5>}
```
Also goes for each occurrence below

File: src/shared/hooks/useDownload.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  const [loading, setLoading] = React.useState(false);
+
+  async function downloadFile(url, name) {
+    setLoading(true);
+
+    const fileName = name || url.split('/').pop();
+
+    fetch(url)
+      .then(response => response.blob())
+      .then(blob => {
```
Comment (janjaap) [inline src/shared/hooks/useDownload.js]: You should also catch any exceptions that occur in the `fetch`

File: src/components/filterPanel/FilterPanel.js (hunk 1 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: removed

File: src/components/filterPanel/FilterPanel.js (hunk 2 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: nice!

File: src/components/filterPanel/FilterPanel.js (hunk 3 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: sure!!

File: src/components/filterPanel/FilterPanel.js (hunk 4 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: indeed, missed that!

File: src/components/filterPanel/FilterPanel.js (hunk 5 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: yes!

File: src/components/filterPanel/FilterPanel.js (hunk 6 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: all right!

File: src/components/filterPanel/FilterPanel.js (hunk 7 above)
Comment (aditudorache) [reply src/components/filterPanel/FilterPanel.js]: done!

File: src/shared/hooks/useDownload.js (hunk 8 above)
Comment (aditudorache) [reply src/shared/hooks/useDownload.js]: done

**Pull request merged**